### ISSUE TITLE
Checksum mismatch | possible race condition on publish?

### ISSUE THREAD

**Issue by andreasrs** [npm/npm#10574]
My newly published package https://www.npmjs.com/package/roc is getting a checksum mismatch.

At first I thought this was a local issue, but doing random downloads using curl and then shasum'ing confirms the problem.

**Sometimes npm returns different content when requesting the same tarball.**

Example curls:

```
21:00:25 andreas@x230 inittest curl -s "https://registry.npmjs.org/roc/-/roc-1.0.0-alpha6.tgz" | shasum
1964f65b1338cafc287d2969339985334f0b5451  -
21:00:28 andreas@x230 inittest curl -s "https://registry.npmjs.org/roc/-/roc-1.0.0-alpha6.tgz" | shasum
1964f65b1338cafc287d2969339985334f0b5451  -
21:00:30 andreas@x230 inittest curl -s "https://registry.npmjs.org/roc/-/roc-1.0.0-alpha6.tgz" | shasum
1964f65b1338cafc287d2969339985334f0b5451  -
21:00:31 andreas@x230 inittest curl -s "https://registry.npmjs.org/roc/-/roc-1.0.0-alpha6.tgz" | shasum
2ef338e7ecea251bb1fe9ce432005239a4ecba02
```

One observation I've made is that my current Travis setup builds and tests two  different Node.js versions in parallel. Travis attempted to publish this version twice. One of the publish attempts failed as expected. 

I don't know if this is relevant, but wanted to mention it just in case the `npm publish` flow is somehow sensitive to race conditions and I have somehow ended up with two different tarballs for the same version distributed in your caches.

**Comment by zkat** [comment]
Closing as a duplicate of https://github.com/npm/npm/issues/14626
